docs/standard/native-interop/tutorial-comwrappers.md

```diff
@@ -0,0 +1,430 @@
+
+```csharp
+IDemoGetType inst = ComInterfaceDispatch.GetInstance<IDemoGetType>((ComInterfaceDispatch*)_this);
+```
+
+Now that you have a C# method that can be inserted into a vtable, you can construct the vtable. Note the use of [`RuntimeHelpers.AllocateTypeAssociatedMemory()`][api_allocatetypeassociatedmemory] for allocating memory in a way that works with [unloadable][doc_unloadability] assemblies.
+
+```csharp
+GetIUnknownImpl(
+    out IntPtr fpQueryInterface,
+    out IntPtr fpAddRef,
+    out IntPtr fpRelease);
+
+{
+    int tableCount = 4;
+    int idx = 0;
+    var vtable = (IntPtr*)RuntimeHelpers.AllocateTypeAssociatedMemory(
+        typeof(DemoComWrappers),
+        IntPtr.Size * tableCount);
+    vtable[idx++] = fpQueryInterface;
+    vtable[idx++] = fpAddRef;
+    vtable[idx++] = fpRelease;
+    vtable[idx++] = (IntPtr)(delegate* unmanaged<IntPtr, IntPtr*, int>)&ABI.
+IDemoGetTypeManagedWrapper.GetString;
+    Debug.Assert(tableCount == idx);
+    s_IDemoGetTypeVTable = (IntPtr)vtable;
+}
+```
+
+The allocation of vtables is the first part of implementing `ComputeVtables()`. You should also construct comprehensive definitions for types that you're planning to support &ndash; think `DemoImpl`. Using the constructed vtables, you can now create a series of [`ComInterfaceEntry`][api_cominterfaceentry] instances that represent the complete view of the managed object in COM.
+
+```csharp
+{
+    s_DemoImplDefinitionLen = 2;
+    int idx = 0;
+    var entries = (ComInterfaceEntry*)RuntimeHelpers.AllocateTypeAssociatedMemory(
+        typeof(DemoComWrappers),
+        sizeof(ComInterfaceEntry) * s_DemoImplDefinitionLen);
+    entries[idx].IID = IDemoGetType.IID_IDemoGetType;
+    entries[idx++].Vtable = s_IDemoGetTypeVTable;
+    entries[idx].IID = IDemoStoreType.IID_IDemoStoreType;
+    entries[idx++].Vtable = s_IDemoStoreVTable;
+    Debug.Assert(s_DemoImplDefinitionLen == idx);
+    s_DemoImplDefinition = entries;
+}
+```
+
+The allocation of vtables and entries for the Managed Object Wrapper can and should be done ahead of time since the data can be used for all instances of the type. The work here could be performed in a `static` constructor or a module initializer, but it should be done ahead of time so the `ComputeVtables()` method is as simple and quick as possible.
+
+```csharp
+protected override unsafe ComInterfaceEntry* ComputeVtables(object obj, CreateComInterfaceFlags flags, 
```

Comment:
> If I implement my own QI - can I just register IUnknown and nothing else here? 

Yep, that is an option. More advanced than even what we have here.

>I assume the runtime provided QI implementation uses the returned table to do its job, right?

Right.

> Is it valid (expected) to return different interface set per-instance (even of the same type)?

That is possible for sure but it would make things difficult to follow and possibly fail in some COM scenarios depending on how the type is exposed.